### ISSUE TITLE
The JS backend generates incorrect code for Agda code that uses reflection

### ISSUE THREAD

**Issue by nad** [agda/agda#5420]
Consider the following code:
```agda
open import Agda.Builtin.IO
open import Agda.Builtin.Reflection
open import Agda.Builtin.String
open import Agda.Builtin.Unit

_>>=_ : {A B : Set} → TC A → (A → TC B) → TC B
_>>=_ = λ x f → bindTC x f

postulate
  putStr : String → IO ⊤

{-# COMPILE JS putStr =
    function (x) {
      return function(cb) {
        process.stdout.write(x); cb(0); }; } #-}

main : IO ⊤
main = putStr "Success\n"
```
If you compile this using the JS backend and run the code using `node`, then you get an error message:
```
[…]/jAgda.Bug.js:9
exports["_>>=_"] = z_jAgda_Agda_Builtin_Reflection["bindTC"](null)(null)(null)(null);
                                                            ^

TypeError: z_jAgda_Agda_Builtin_Reflection.bindTC is not a function
    at Object.<anonymous> ([…]/jAgda.Bug.js:9:61)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
```
It should be possible to compile and run code that uses reflection, if reflection primitives are only used on the meta-level.

**Comment by asr** [comment]
What about of cherry-picking the fix into `release-2.6.2`?

**Comment by nad** [comment]
I have no strong opinion on this.
